index.dev.js

```diff
@@ -25,56 +26,59 @@ function runInsect(fmt, line) {
   return res;
 }
 
-// Handle command line arguments
-if (process.argv.length >= 4) {
-  usage();
-} else if (process.argv.length == 3) {
-  var arg = process.argv[2];
-  if (arg === "-h" || arg === "--help") {
+// Top-level await is not supported in Node 12 and earlier.
+(async function() {
+  if (process.env.INSECT_NO_RC !== "true") {
+    function processRcLine(line) {
+      var res = runInsect(Insect.fmtPlain, line);
+      // We really only care when it breaks
+      if (res && res.msgType === "error") {
+        console.error(res.msg);
+        process.exit(1);
+      }
+    }
+
+    var [util, lineReader] = await Promise.all([import("util"), import("line-reader")]);
+    var eachLine = util.promisify(lineReader.eachLine);
+
+    try {
+      await eachLine(path.join(xdgBasedir.config, "insect/insectrc"), processRcLine);
+    } catch (err) {
+      if (err.code === "ENOENT") {
+        try {
+          var os = await import("os");
+          await eachLine(path.join(os.homedir(), ".insectrc"), processRcLine);
+        } catch (err) {
+          if (err.code !== "ENOENT") {
+            throw err;
+          }
+        }
+      } else {
+        throw err;
+      }
+    }
```

Comment:
Can we avoid some code duplication here by constructing a list with possible rc-file locations, and then take the first one that exists?

Reply:
Good suggestion, I went ahead and did just that.